batch_job/src/work_queue_factory.c

```diff
@@ -108,6 +107,10 @@ static int64_t factory_timeout = 0;
 
 struct batch_queue *queue = 0;
 
+static const char *mesos_master = NULL;
```

Comment:
I agree. For clarity, you are creating this variable and passing it to the batch_job system. The variable from makeflow's perspective is not important, it only means something in batch_job. The others that you mentioned should also be cleaned up, though I am not relying on you for that.